### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Kubik42**

### CODE REVIEW

**Comment by yeekangc** [comment]
@lauracowen, if you have time, can you please review the changes here?  You have provided feedback previously and are looking at improvements for what we have here too.

File: README.adoc [hunk 1]
```diff
@@ -24,23 +24,28 @@ Learn how to use JAX-RS, CDI, and JSON-P to build a MicroProfile application.
...
-and help manage the lifecycle of your application.
-
-You will create an application that provides an `inventory` service, which allows you to retrieve
-the information for a particular host or a list of all previously registered hosts.
-If the requested host is not in the inventory, the `inventory` service attempts to retrieve
-that host's information by calling its `system` service.
-
-For example, you will be able to access the following URL to retrieve a list of all hosts in the inventory:
-
+In this guide, you will learn how to write RESTful web services using JAX-RS and JSON-P while utilizing
```
Comment (yeekangc) [inline README.adoc]: What the rationale to remove the line that explains MicroProfile?

Comment (Kubik42) [reply README.adoc]: it was weird for the title to be "What you'll learn" and then immediately go into explaining what microprofile is. I wanted to move that section somewhere else, or lower at least, I guess I just forgot. Anyway, we also always started our guides with "In this guide, you will learn..."

**Comment by yeekangc** [comment]
@evelinec, given the scope of the changes, we need someone to re-test everything.

File: README.adoc (hunk 1 above)
Comment (Kubik42) [reply README.adoc]: fixed

Comment (lauracowen) [reply README.adoc]: Can you remove 'In this guide' (it's not necessary; just say 'You will learn...').

I agree that it should be re-reviewed/tested if it's significantly changed.

I was discussing this guide with @NottyCode  and @gcharters earlier in the week after reviewing part of it myself and we were concerned about the length of it and that the text needs a tighter focus to be about integrating JAX-RS with CDI but without some of the duplication of the JAX-RS and Maven guides. I need to look at it with @gcharters though to know just what should be updated (as he knows the area better and what he and Alasdair intended).

**Comment by Kubik42** [comment]
I guess we could have the REST guide as a prerequisite to this one. Otherwise, we can strip some of the maven blocks from the test section

**Comment by Kubik42** [comment]
@yeekangc @lauracowen My latest attempt to clear things up based on our latest discussion.

I was able to significantly shorten the adoc; from ~540 lines to ~350 (the rendered guide will be even shorter). Like we talked, I attempted to heavily focus the guide on MicroProfile, explaining what it is, what technologies its made up of, as well as where and how to use these technologies in the app. As a result, I removed the section on CDI since it seems to confuse the focus of the guide. I even added in a few comments in the imports so its clear which imports are JAX-RS, which are CDI, and which are JSON-P. I also chose to omit all utility classes for the guide since they are not that important (I did add detailed comments to these so the users can read them).

I went with Laura's suggestions: the code snippets are no longer split up (except for the test one). As a result, its now much easier to build the classes form scratch (I tested this myself and it takes under a minute to setup everything up). Like suggested as well, I added a brief section that talks about trying the app first before building it. Finally, I also added a diagram in the starting section (as of Dec 28, this part is not yet commited).

**Comment by Kubik42** [comment]
Closing this since mp-intro is no longer relevant. We should open a new review issue when the cdi guide is ready to be published.

**Pull request closed without merging**